### ISSUE TITLE
'GetConsoleScreenBufferInfo' failed on git bash

### ISSUE THREAD

**Issue by LiShanwenGit** [mintty/mintty#1188]
I get the terminal size of git bash shell, but the function return error code, like this:
```
   CONSOLE_SCREEN_BUFFER_INFO csbi;
    switch (1) {
    case 0: nhandle = STD_INPUT_HANDLE;
        break;
    case 1: nhandle = STD_OUTPUT_HANDLE;
        break;
    case 2: nhandle = STD_ERROR_HANDLE;
        break;
    }
    handle = GetStdHandle(nhandle);
    if (handle == NULL)
        printf("handle cannot be retrieved");
    if (handle == INVALID_HANDLE_VALUE)
        printf("handle is invalid\n");
    if (GetConsoleScreenBufferInfo(handle, &csbi) !=0)
        printf("Error: get info failed\n"); 
```
the result print Error: get info failed.
so, I could not know why, can you help me?, thank you a lot.  : )

**Comment by LiShanwenGit** [comment]
the compiler is: 
$ gcc -v
Using built-in specs.
COLLECT_GCC=D:\MSYS2\mingw64\bin\gcc.exe
COLLECT_LTO_WRAPPER=D:/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../gcc-12.2.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mi
ngw64/include --libexecdir=/mingw64/lib --enable-bootstrap --enable-checking=release --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,fortran,ada,objc,obj-c++,jit --enable-shared --enable-sta
tic --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-filesystem-ts --enable-libstdcxx-time --disable-libstdcxx-pch --enable-lto --enable-libgomp --dis
able-multilib --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl
=/mingw64 --with-pkgversion='Rev6, Built by MSYS2 project' --with-bugurl=https://github.com/msys2/MINGW-packages/issues --with-gnu-as --with-gnu-ld --disable-libstdcxx-debug --with-boot-ldflags=-static-libstdc+
+ --with-stage1-ldflags=-static-libstdc++
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (Rev6, Built by MSYS2 project)

**Comment by LiShanwenGit** [comment]
![image](https://user-images.githubusercontent.com/35484039/204973402-cd0eeca1-34f9-450a-af20-0520fe4c926e.png)
the bash is ok, but the git-bash is bad, : (

**Comment by k-takata** [comment]
You might want to see this:
https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs

**Comment by LiShanwenGit** [comment]
> You might want to see this: https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs

Great, it's good, thank you : )
solution: 
1. create a 'git-bash.config' file in <git installed dir>/etc
2. add 'MSYS=enable_pcon' into git-bash.config file
3. restart git-bash shell

**Comment by mintty** [comment]
Thanks, added this option to the wiki.

**Comment by Biswa96** [comment]
FYI, msys2 enables conpty by-default from this date https://www.msys2.org/news/#2022-09-24-conpty-support-enabled-by-default

**Comment by mintty** [comment]
Thanks, noted in the wiki.

**Comment by LiShanwenGit** [comment]
Hi, gays, Thank you very much for your help. However, to my surprise, the performance difference between the two is so large. When git bash add "MSYS=enable_pcon", its terminal output efficiency decreases greatly. Ithink that the terminal of Windows really needs to be improved.

**Comment by mintty** [comment]
We're talking here about mintty, not Windows terminal, right?
And what does `uname -a` say for you?

**Comment by LiShanwenGit** [comment]
Yes, you are right, this is not a problem with MITTY, but a problem with Windows, just to mention this.
thanks.
